src/sync_manager.hpp

```diff
@@ -58,39 +66,32 @@ friend struct SyncSession;
     void set_log_level(util::Logger::Level) noexcept;
     void set_logger_factory(SyncLoggerFactory&) noexcept;
     void set_error_handler(std::function<sync::Client::ErrorHandler>);
-    void set_login_function(SyncLoginFunction);
+    void set_bind_session_function(SyncBindSessionFunction);
 
     /// Control whether the sync client attempts to reconnect immediately. Only set this to `true` for testing purposes.
     void set_client_should_reconnect_immediately(bool reconnect_immediately);
     /// Control whether the sync client validates SSL certificates. Should *always* be `true` in production use.
     void set_client_should_validate_ssl(bool validate_ssl);
 
-    std::shared_ptr<SyncSession> get_session(const std::string& path, const SyncConfig& config);
-    std::shared_ptr<SyncSession> get_existing_active_session(const std::string& path) const;
+    // Get a sync user for a given identity, or create one if none exists yet, and set its token.
+    // If a logged-out user exists, it will marked as logged back in.
+    shared_ptr<SyncUser> get_user(std::string identity, std::string user_token, bool is_admin=false);
+    // Get an existing, valid sync user for a given identity.
+    shared_ptr<SyncUser> get_existing_valid_user(std::string identity);
```

Comment:
You're right, I'm not sure `valid` means anything in this context. It might make more sense if this method was named `get_existing_active_user` or `get_existing_logged_in_user`.